### ISSUE TITLE
Odd Basement GroundDomain Error

### ISSUE THREAD

**Issue by Myoldmopar** [NREL/EnergyPlus#5073]
@mmatts A user found that EnergyPlus was failing using the Site:GroundDomain:Basement object in 8.3.0.  

It failed with `FATAL:GetPipingSystemsInput: Preceding input errors cause program termination.` because it said there were blank inputs found.  The user was not using a piping system, so we may have to re-work the error message text since that code has been generalized to slab/basement.
- [ ] Fix error message text

I debugged in and found it was hitting that error condition [here](https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/PlantPipingSystemsManager.cc#L1603), because A8 was blank.  A8 is the `Basement Wall Vertical Insulation Material Name`, and the user had A7 (`Vertical Insulation`) set to No, so they did not expect that to be a required field.
- [ ] Properly check erroneous/valid conditions on basement inputs

I don't know about distributing the idf just yet, so in the meantime, this should be reproducible by turning vertical insulation to No, and having a blank material name in A8.

As a slight tangent to this, the user was a bit confused about the OtherSideConditions keyword that they had to use to hook the ground domains to the surfaces, so that could use a rename.  Which would require transition that I could take care of.
- [ ] Change keyword for ground domain on OSC models
- [ ] Transition rules for keyword change

**Comment by Myoldmopar** [comment]
@dstuder

**Comment by mjwitte** [comment]
@Myoldmopar @mmatts If you decide to proceed with a change to the OSC referencing, I'd rather see the OSC layer removed from the input and allow the ground-domain to be directly referenced in the surface object.  The current referencing is a bit obtuse.

```
SurfaceProperty:OtherSideConditionsModel,
    GroundCoupledOSCM, !- Name
    GroundCoupledSurface; !- Type of Modeling   

  BuildingSurface:Detailed,
    F1-1,                    !- Name
    FLOOR,                   !- Surface Type
    FLOOR-SLAB-1,            !- Construction Name
    SPACE1-1,                !- Zone Name
    OtherSideConditionsModel,  !- Outside Boundary Condition
    GroundCoupledOSCM,     !- Outside Boundary Condition Object

Site:GroundDomain:Slab,
    IngradeCoupledSlab,      !- Name
<snip>
    1,                       !- Evapotranspiration Ground Cover Parameter
    GroundCoupledOSCM,       !- Slab Boundary Condition Model Name

```

It would seem to be more direct to reference the grounddomain object directly in the surface.

```
  BuildingSurface:Detailed,
    F1-1,                    !- Name
    FLOOR,                   !- Surface Type
    FLOOR-SLAB-1,            !- Construction Name
    SPACE1-1,                !- Zone Name
    **GroundDomainSlab**,  !- Outside Boundary Condition
    **IngradeCoupledSlab**,     !- Outside Boundary Condition Object

Site:GroundDomain:Slab,
    IngradeCoupledSlab,      !- Name
<snip>
    1,                       !- Evapotranspiration Ground Cover Parameter
   DELETE --> GroundCoupledOSCM,       !- Slab Boundary Condition Model Name
```

**Comment by mjwitte** [comment]
@Myoldmopar Transition8-2-0-to-8-3-0 did some mysterious things with Site:GroundDomain:Slab.  Look at the field names that are in example file 5ZoneAirCooledWithCoupledInGradeSlab

```
  Site:GroundDomain:Slab,
    IngradeCoupledSlab,      !- Name
    5,                       !- Month
    1,                       !- Day of Month
    5,                       !- Day Type
    1.8,                     !- Maximum Dry-Bulb Temperature {C}
    3200,                    !- Daily Dry-Bulb Temperature Range {deltaC}
    836,                     !- Dry-Bulb Temperature Range Modifier Type
    30,                      !- Dry-Bulb Temperature Range Modifier Day Schedule Name
    50,                      !- Humidity Condition Type
    15.5,                    !- Wetbulb or DewPoint at Maximum Dry-Bulb {C}
    12.8,                    !- Humidity Condition Day Schedule Name
    17.3,                    !- Humidity Ratio at Maximum Dry-Bulb {kgWater/kgDryAir}
    1,                       !- Enthalpy at Maximum Dry-Bulb {J/kg}
    GroundCoupledOSCM,       !- Daily Wet-Bulb Temperature Range {deltaC}
    InGrade,                 !- Barometric Pressure {Pa}
    Slab Material-In-grade,  !- Wind Speed {m/s}
    Yes,                     !- Wind Direction {deg}
    Slab Insulation,         !- Rain Indicator
    Full,                    !- Snow Indicator
    ,                        !- Daylight Saving Time Indicator
    Yes,                     !- Solar Model Indicator
    Slab Insulation,         !- Beam Solar Day Schedule Name
    2,                       !- Diffuse Solar Day Schedule Name
    Timestep;                !- ASHRAE Clear Sky Optical Depth for Beam Irradiance (taub) {dimensionless}
```

**Comment by Myoldmopar** [comment]
Sounds like a step too far to get this bug fixed for the 8.4.0 release, but I agree about stripping out the other side conditions layer from userland.

**Comment by Myoldmopar** [comment]
AAAAHHH What?  I know what happened.  I'll fix it separately.

**Comment by mjwitte** [comment]
@Myoldmopar @mmatts Removing the OSC layer could be done as part of the integration of the groundtemp calc utility, right?  'cause that will cause transitions to remove the K-A fields?

**Comment by Myoldmopar** [comment]
The problem is deeper than that because the other side conditions model layer sets up the communication between domains, including data structures, input validation.  It's not _just_ the name reference, there's other stuff behind the scenes.

Again, I'd love to see it go, but I think it should be separate.

**Comment by mjwitte** [comment]
The internal mechanism could stay there and just be invisible to the user.  But maybe that's asking too much.

**Comment by Myoldmopar** [comment]
We are in agreement.  The behind the scenes work would definitely stay.  But I feel strongly that task should stand alone as a cleanup/usability issue in FY16.

**Comment by mjwitte** [comment]
@mitchute Can the initial problem here be fixed for v8.5.0 (without any changes to the OSC stuff that's discussed in here)?

**Comment by mitchute** [comment]
@mjwitte @Myoldmopar 
Aside from the OSM stuff, it looks like there are two other tasks.
- update error message
- improve checks to validate input

I should be able to do that in the next day or two.

**Comment by Myoldmopar** [comment]
Closed via #5548
